2020/06/02/why-upgrading-software-libraries-is-imperative.html.md

```diff
@@ -0,0 +1,44 @@
+---
+author: "Selvakumar Arumugam"
+title: "Why Upgrading Software Libraries is Imperative"
+tags: software, version, upgrade
+gh_issue_number: 
+---
+
+Stack applications primarily consist of the Operating System as well as front and backend programming languages including their necessary list of library packages. The Operating System and programming language software can be periodically updated and run on the latest version. However, what about the long list of libraries being used in the application’s front and backend? As we know, It can be quite daunting to maintain and individually update a long list of software libraries. Still, it is important to keep them updated in several ways while mitigating the level of complexity. This post dives into our experience upgrading a complex stack with a list of libraries. What you will need, how to go about it, and the benefits of such an activity are all examined below. 
+
+The application stack in question contained decade-old software and included extensive libraries when we received it from our client. The application consisted of languages including Java, Scala, Kotlin and JavaScript in tandem with an extensive list of libraries. The initial plan was to upgrade the complete stack all at once due to the gap between versions. This proved to be more difficult than expected due to a host of deprecated and removed functionalities in addition to their inter-dependency on a few of the libraries. 
```

Comment:
I think it could be helpful to give an example of what "libraries" are. Developers know this, but this is also targeted at business people who need to understand the importance of paying for it (in time, money).

So showing a list of libraries for an app (maybe Ridgetop's) and explaining what they each do might help business people understand what we mean, and how most applications are built of many pieces written by many people, each separately maintained (or not maintained anymore!).

I would list the operating system distribution, web server, and database server too, which are all big pieces. Explaining what each part does and giving names can illustrate.

Reply:
@jonjensen 
I have added description of libraries to list to the list of Ridgetop libraries.  Please review and Shall we link this spread sheet to the blog post.

https://docs.google.com/spreadsheets/d/1k8W7pR9V48K8zHXno4sWHfSb8FGRxkANXOGZteARW9g/edit#gid=362307797

Reply:
That's the kind of thing it will help to see, yes. I don't think we want to link to an external resource, but we can keep just the most important parts of that and turn it into an HTML table. I'll have one of our editors do that. Thanks!

Reply:
@jonjensen I see the libraries list is missing in the blog post as HTML table as you stated above. Just wanted to point out in case if it is missed by mistake. Thanks.
cc: @obnoxiousorc

Reply:
@selvait90 Sorry we missed that. We'll get it in there ASAP.

Reply:
@selvait90 I just added those. Apologies for missing it earlier!

Reply:
No problem, It was one of comment and didn't highlighted well to notice at the time of publish. Thank you so much @jonjensen & @obnoxiousorc